<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda_softhand">

  <!-- MODELS -->
  <xacro:include filename="$(find franka_description)/robots/panda_arm.urdf.xacro"/>
  <xacro:include filename="$(find soft_hand_description)/model/soft_hand_soma_panda.urdf.xacro"/>

  <link name="world"/>

  <!-- PANDA ARM -->
  <xacro:panda_arm name="panda_arm" parent="world">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
  </xacro:panda_arm>

  <!-- RIGHT SOFTHAND -->
  <xacro:soft_hand_soma_panda parent="panda_link8" name="right_hand" left="false" withAdaptiveTransmission="true" useMimicTag="$(arg use_mimic_tag)">
    <origin xyz="0 0 0" rpy="0 0 3.1415"/>
  </xacro:soft_hand_soma_panda>

  <!-- EXTRA ENDEFFECTOR -->
  <joint name="right_hand_ee_joint" type="fixed">
    <origin xyz="0.002 0.00695 0.1" rpy="0 -1.5708 3.1415"/>
    <parent link="right_hand_panda_flange" />
    <child link="right_hand_ee_link" />
  </joint>

  <link name="right_hand_ee_link">
  </link>

</robot>
